### ISSUE TITLE
Install git package under `lib` rather than `src`

### ISSUE THREAD

**Issue by petobens** [python-poetry/poetry#2217]
Given the following `pyproject.toml` file:
```toml
[tool.poetry]
name = "pbpp-poetry"
version = "0.1.0"
description = ""
authors = ["petobens <foo@yahoo.com>"]

[tool.poetry.dependencies]
python = "^3.8"
pdbpp = {git = "https://github.com/pdbpp/pdbpp"}

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```
If I run `poetry install` this will indeed install the pdbpp master branch but if I run `poetry run python foo.py` on the following `foo.py` file:
```python
x = 2
breakpoint()
print(x)
y = 3
``` 
 then pdbpp is not activated (it falls back to the standard pdb). I guess this is because poetry installs the  gitpackage in development or "src" mode since `poetry run pip list` shows:
```
Package        Version                Location
-------------- ---------------------- -------------------------------------------------------------------------------------
fancycompleter 0.9.2.dev2-g8ae963e    /home/pedro/.cache/pypoetry/virtualenvs/pbpp-poetry-uIeetswW-py3.8/src/fancycompleter
pdbpp          0.10.3.dev170+g5eefc46 /home/pedro/.cache/pypoetry/virtualenvs/pbpp-poetry-uIeetswW-py3.8/src/pdbpp/src
pip            19.2.3
Pygments       2.6.1
pyrepl         0.8.5.dev44            /home/pedro/.cache/pypoetry/virtualenvs/pbpp-poetry-uIeetswW-py3.8/src/pyrepl
setuptools     41.2.0
six            1.14.0
wmctrl         0.3
```
On the other hand if I uninstall pdbpp git with `poetry remove pdbpp` and install the pypi version with `poetry add pbpp` then pdbpp works just fine since it is installed as a "library":
```

-------------- ------- ----------------------------------------------------------------------------------------------
fancycompleter 0.9.1   /home/pedro/.cache/pypoetry/virtualenvs/pbpp-poetry-uIeetswW-py3.8/lib/python3.8/site-packages
pdbpp          0.10.2  /home/pedro/.cache/pypoetry/virtualenvs/pbpp-poetry-uIeetswW-py3.8/lib/python3.8/site-packages
pip            19.2.3
Pygments       2.6.1
pyrepl         0.9.0   /home/pedro/.cache/pypoetry/virtualenvs/pbpp-poetry-uIeetswW-py3.8/lib/python3.8/site-packages
setuptools     41.2.0
wmctrl         0.3
```
So my question is: is it possible to install the git package in a way that it works as the normal pypi package?

Note: I've also asked this in the pdpp repo [here](https://github.com/pdbpp/pdbpp/issues/122#issuecomment-602128595) but I think it might be a poetry issue (or feature) rather than a problem with pdbpp.

**Comment by blueyed** [comment]
> then pdbpp is not activated (it falls back to the standard pdb)

Note that this normally means that its `.pth` file is not installed (which hijacks the "pdb" module).  This gets installed via `data_files` (https://github.com/blueyed/pdbpp/blob/aba2744696f1ec05e25d48cf1fdfc711f27e2b02/setup.py#L68-L73).
(I've played around a bit with doing this also for develop-mode/editable installs, but it is not trivial, since pip does not support uninstalling extra files then.)

**Comment by finswimmer** [comment]
Hello @petobens ,

installing git dependencies in non-development mode is currently not supported. But there is a PR for it waiting for a review: https://github.com/python-poetry/poetry/pull/1810

fin swimmer

**Comment by petobens** [comment]
I see. Thanks both for the explanations.

**Comment by dazza-codes** [comment]
nit: why is this issue closed when the MR is not merged yet?

**Comment by petobens** [comment]
Hi @finswimmer and @blueyed. Now that the pr to allow non-development mode has been merged as per https://github.com/python-poetry/poetry-core/pull/23/files I tried adding the following line to my `pyproject.toml`
```
pdbpp = {git = "https://github.com/pdbpp/pdbpp.git", develop = false}
```
and did a `poetry update` (or even a clean `poetry install`) but pdpp still seems to beinstalled in dev mode right?
```
$ poetry run pip list | grep pdbpp
pdbpp             0.10.3.dev197+gdea1c51 /home/pedro/.cache/pypoetry/virtualenvs/test-precommit-vTiR5TE--py3.8/src/pdbpp/src
```
Am i doing something wrong or this is expected?
